src/services/DFGVariable.jl

```diff
@@ -162,14 +162,14 @@ Since the `timestamp` field is not mutable `setTimestamp` returns a new variable
 Use [`updateVariable!`](@ref) on the returened variable to update it in the factor graph if needed. Alternatively use [`setTimestamp!`](@ref).
 See issue #315.
 """
-function setTimestamp(v::DFGVariable, ts::DateTime; verbose::Bool=true)
+function setTimestamp(v::DFGVariable, ts::ZonedDateTime; verbose::Bool=true)
```

Comment:
WOW, this is a major breaking change!  This will make the entire ecosystem have to drop `DateTime` which is not reasonable I think -- not sure if we should resolve this with a conversion (think Julia will auto-promote):
```
convert(::Type{ZonedDateTime}, dt::DateTime)
```
so that when going into and out of Neo4j it does the conversion and then we can work on dispatch to support both cases `DateTime` and `ZonedDateTime`?

Reply:
For v0.9.0 we would have to do all the deprecation warnings etc.

Reply:
I'll internalize the CGDFG changes with ZonedDateTime, I was hoping it was a small set of changes, but it would go all the way to IIF. Rather, from now on DateTime is UTC and we'll handle ZonedDateTime inside CGDFG only. Changes on their way.

Reply:
two problems ecosystem wide, Timezone and Nanosecond accuracy.  Shall we do those together in future PRs?  Internalize in CGDFG for now and expand from there?

Reply:
In case you haven't seen it, try `now(UTC)`, we can add that to all the default constructors. That way we always store DateTime in UTC, as you suggest, and then the functions on top of that can get the local time if needed.

That being said, It might be worthwhile to make the change to add Location. I think we can use `DateTime(zdt, Local)`
Maybe leave it as a separate PR?
I guess we can assume local time? 

@dehann, the `convert(::Type{ZonedDateTime}, dt::DateTime)` will also have to assume the TimeZone, but it is not lossless, so I would vote for using the constructor over converter.

Reply:
see #523

Reply:
:+1: Updated in IIF - https://github.com/JuliaRobotics/IncrementalInference.jl/pull/762/files

Reply:
eg #582